charm/kafka/actions/upgrade-kafka

```diff
@@ -0,0 +1,12 @@
+#!/usr/local/sbin/charm-env python3
+
+import kafkautils
+import subprocess
+
+from charms.reactive import remove_state
+
+
+remove_state('kafka.nrpe_helper.installed')
+remove_state('kafka.started')
```

Comment:
This will trigger a call to the `configure_kafka` handler in `kafka.py`, won't it?
Shouldn't you also remove `snap.installed.kafka` so it downloads the new version before running that handler?

Reply:
I found it necessary to directly trigger the snap upgrade from the action. Pending reactive handlers did not seem to fire in practice, and I didn't want to wait for the next update-status hook to run.